src/ngx_http_lua_bodyfilterby.c

```diff
@@ -140,6 +146,19 @@ ngx_http_lua_body_filter_by_chunk(lua_State *L, ngx_http_request_t *r,
 
     lua_settop(L, 0);
 
+    if (ctx->body_exited) {
+        /* send last content */
+        for (cl = in; cl->next; cl = cl->next) {
+            /* do nothing */
+        }
+        /* make ngx_http_write_filter_module happy */
+        cl->buf->last_buf = 1;
+
+        ctx->eof = 1;
+
+        return NGX_OK;
```

Comment:
@rainingmaster Your use case looks wrong from the perspective of the HTTP protocol. You are essentially truncating the original response body but still pretend to the client to be well formed. This is just wrong. The client should be notified of this data truncation instead of being cheated. What exactly is your real world business-level use case?